noether_tpp/core/include/noether_tpp/core/tool_path_planner_pipeline.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * @file pipeline.h
+ * @copyright Copyright (c) 2021, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <noether_tpp/core/mesh_modifier.h>
+#include <noether_tpp/core/tool_path_modifier.h>
+#include <noether_tpp/core/tool_path_planner.h>
+
+#include <pcl/PolygonMesh.h>
+
+namespace noether_tpp
+{
+/**
+ * @brief The ToolPathPlannerPipeline class - Collects together a set of mesh modifiers, a planner,
+ * and a set of tool path modifiers. These sub-units then can be called using a single line to
+ * pre-process meshes, plan paths, and operate on the paths.
+ */
+class ToolPathPlannerPipeline
+{
+public:
+  ToolPathPlannerPipeline(std::unique_ptr<MeshModifier>&& mesh_mod,
+                          std::unique_ptr<ToolPathPlanner>&& planner,
+                          std::unique_ptr<ToolPathModifier>&& tool_path_mod);
```

Comment:
To me this style of construction puts a lot of burden on the user who has to create all these things ahead of time. Consider most people probably just want to create a pipeline directly and maybe change a few of defaults. Obviously we can keep this full constructor but other alternatives would nice too.

Reply:
Would an alternative constructor be one where we give default values for the modifiers and planners, or are you thinking of something different?

Reply:
@jdlangs The ToolPathPlannerPipeline is one of the less-well-defined concepts in this redesign.  Perhaps you could join us for a discussion about what you would want to see as a use case for it?

@marip8 can correct me here, but I believe the intention was to offer a few 'specified' or 'standard' pipelines that come pre-filled and you just configure it, as well as the 'base' or 'empty' pipeline that you have to custom-construct yourself.

(I don't necessarily disagree that this constructor looks intimidating and like a massive pain.)

Reply:
I'm thinking alternatives would be either directly passing parameters like you mentioned, Michael, or just allowing a default construction and letting the user modify particular pieces as they see fit. But I don't think anything needs to change right now, and I'll look forward to future discussions.

Reply:
We'll keep this in mind and add other constructors once we have more implementations of the modifiers and planners to provide.